sky/serve/autoscalers.py

```diff
@@ -264,30 +267,27 @@ def _set_target_num_replica_with_hysteresis(self) -> None:
             f'{self.scale_down_consecutive_periods}')
 
     @classmethod
-    def _select_replicas_to_scale_down(
+    def _select_nonterminal_replicas_to_scale_down(
             cls, num_limit: int,
             replica_infos: Iterable['replica_managers.ReplicaInfo']
     ) -> List[int]:
-
         status_order = serve_state.ReplicaStatus.scale_down_decision_order()
-        # Also sort by provisioned time (indicated by replica_id) to make sure
-        # we terminate the replicas that starts provisioning later first
-        replica_infos_sorted = sorted(
+        assert all(info.status in status_order for info in replica_infos), (
+            'All replicas to scale down should be in provisioning or launched '
+            'status.', replica_infos)
+        replicas = sorted(
             replica_infos,
             key=lambda info: (
-                status_order.index(info.status)
-                # Use -1 for other status that will be terminated first.
-                # Including: NOT_READY, SHUTTING_DOWN, FAILED,
-                # FAILED_CLEANUP, PREEMPTED, UNKNOWN.
-                if info.status in status_order else -1,
-                # `-info.replica_id` is to furhter sort the replicas with the
-                # same state by the time it starts launching. In that case,
-                # if two replicas are both in PROVISIONING state, we will scale
-                # down the one that starts provisioning later, i.e., it will
-                # take a longer time for it to be READY.
+                status_order.index(info.status),
+                # Sort by version in ascending order, so we scale down the older
+                # versions first.
```

Comment:
Q: Can there be more than 2 versions simultaneously? Two consecutive `serve update` with a short gap?